clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -972,16 +975,28 @@ private Record nextFetchedRecord() {
         private List<ConsumerRecord<K, V>> fetchRecords(int maxRecords) {
             if (isFetched)
                 return Collections.emptyList();
+            if (nextInlineException != null) {
```

Comment:
@hachikuji Not sure I fully understand the behavior you described. Say we have a batch with message offset 0, 1, 2, 6 inside it (from a log compacted topic). Message 2 is a bad message.
The behavior with this patch would be:
1. first poll() returns message 0,1, position updated to 2.
2. second poll() throws an exception and the position is still at 2.
3. third poll() skips message 2 and delivers message 6, position becomes 7.

It seems we are not breaking the semantic that the returned record's offset is greater than or equals to the position, right?

From user's perspective, the second poll throws an exception and user can check the position and knows that position 2 is where the bad message is.

The third poll() call skips message 2 and delivers message 3. It seems intuitive because a bad message is technically not available and user ignored the thrown exception by just call poll() again. In that case, offset 3 is the first available message after the current position 2.

Am I missing something?